internal/cluster/distribution/eks/eksprovider/workflow/update_asg_activity.go

```diff
@@ -307,6 +308,18 @@ func (a *UpdateAsgActivity) Execute(ctx context.Context, input UpdateAsgActivity
 		},
 	}
 
+	if input.UseInstanceStore != nil {
+		stackParams = append(stackParams, &cloudformation.Parameter{
+			ParameterKey:   aws.String("UseInstanceStore"),
+			ParameterValue: aws.String(strconv.FormatBool(*input.UseInstanceStore)),
+		})
+	} else if !input.CurrentTemplateVersion.IsLessThan("2.2.0") {
+		stackParams = append(stackParams, &cloudformation.Parameter{
+			ParameterKey:     aws.String("UseInstanceStore"),
+			UsePreviousValue: aws.Bool(true),
+		})
+	}
+
```

Comment:
**Recommended**: for readability I would inline this parameter creation as well (to keep the parameters together at one place).

```golang
{
    // ...
    sdkCloudformation.NewOptionalStackParameter(
        "UseInstanceStore",
        input.UseInstanceStore != nil || input.CurrentTemplateVersion.IsLessThan("2.2.0"),
        strconv.FormatBool(aws.BoolValue(input.UseInstanceStore)), // Note: false default value for old stacks.
    ),
    // ...
}

```

Same at:
https://github.com/banzaicloud/pipeline/pull/3388/files#diff-f41acaa01845aedf481af7231226d88a91a3fb243c7194ee58f375567faeb564R243-R254 (internal/cluster/distribution/eks/eksworkflow/activity_update_node_group.go : 243-254)

Reply:
fixed